### PR TITLE
Use dataflow 3.6.0.

### PR DESCRIPTION
**Pull request by xingweitian**

### CODE REVIEW

File: check_api/src/main/java/com/google/errorprone/dataflow/DataFlow.java [hunk 1]
```diff
@@ -103,7 +106,8 @@ public ControlFlowGraph load(CfgParams key) {
                     }
                   }
                   if (methodPath.getLeaf() instanceof LambdaExpressionTree) {
-                    ast = new UnderlyingAST.CFGLambda((LambdaExpressionTree) methodPath.getLeaf());
+                    MethodTree method = TreeUtils.enclosingMethod(methodPath);
```
Comment (wmdietl) [inline check_api/src/main/java/com/google/errorprone/dataflow/DataFlow.java:109]: Add to for-loop above.

File: check_api/src/main/java/com/google/errorprone/dataflow/DataFlow.java [hunk 2]
```diff
@@ -96,16 +99,21 @@ public ControlFlowGraph load(CfgParams key) {
...
                   ClassTree classTree = null;
+                  MethodTree method = null;
                   for (Tree parent : methodPath) {
                     if (parent instanceof ClassTree) {
                       classTree = (ClassTree) parent;
                       break;
                     }
+                    if (parent instanceof MethodTree) {
+                      method = (MethodTree) parent;
+                      break;
```
Comment (wmdietl) [inline check_api/src/main/java/com/google/errorprone/dataflow/DataFlow.java:110]: Remove the break? You want to go on up and also find the class.

File: check_api/src/main/java/com/google/errorprone/dataflow/DataFlow.java [hunk 3]
```diff
@@ -96,16 +99,21 @@ public ControlFlowGraph load(CfgParams key) {
                   final TreePath methodPath = key.methodPath();
                   final UnderlyingAST ast;
                   ClassTree classTree = null;
+                  MethodTree method = null;
```
Comment (wmdietl) [inline check_api/src/main/java/com/google/errorprone/dataflow/DataFlow.java:102]: For consistency with `classTree`, how about `methodTree`?

**Comment by xingweitian** [comment]
This PR is moved to https://github.com/google/error-prone/pull/1771 and can be closed.

**Pull request closed without merging**